semgrep/tests/e2e/test_dependency_aware_rules.py

```diff
@@ -31,3 +37,23 @@ def test_dependency_aware_rules(run_semgrep_in_tmp_no_symlink, snapshot, rule, t
         run_semgrep_in_tmp_no_symlink(rule, target_name=target)[0],
         "results.json",
     )
+
+
+# Quite awkward. To test that we can handle a target whose toplevel parent
+# contains no lockfiles for the language in our rule, we need to _not_ pass in
+# a target that begins with "targets", as that dir contains every kind of lockfile
```

Comment:
Why does `targets` contain every kind of lockfile?

Reply:
Because of `targets/dependency_aware`

Reply:
Doesnt SCA only ever move up the tree anyway so running semgrep in `targets/basic` is not affected by lock files in `targets/dependency_aware`?

Reply:
Not exactly, though perhaps it ought to work that way. Currently we go to the top-most parent of the first semgrep target, then recursively traverse it for all lockfiles for the languages in the rules. Target files get associated to the nearest lockfile up the tree, but this bug occurred when there were no lockfiles _at all_, not just not lockfiles associated to the target.

Reply:
ah i see

Reply:
I guess that functionality is out of scope of this PR